lib/concurrent/atomic/atomic_fixnum.rb

```diff
@@ -74,10 +68,7 @@ def value=(value)
     #
     #   @return [Fixnum] the current value after incrementation
     def increment
-      @mutex.lock
-      @value += 1
-    ensure
-      @mutex.unlock
+      synchronize { @value += 1 }
```

Comment:
I think the reason it was called `AtomicFixnum` is so that implementations in Java and C can use high performance primitives, which on those platforms do not support overflow.